website/content/en/docs/golang/references/markers.md

```diff
@@ -8,11 +8,153 @@ This document describes [code markers][markers] supported by the SDK.
+	type MemcachedSpec struct {
+		// Size is the size of the memcached deployment. <-- This will become Size's specDescriptors.description.
+		// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Number of pods",xDescriptors="urn:alm:descriptor:com.tectonic.ui:podCount,urn:alm:descriptor:io.kubernetes:custom"
+		Size int32 `json:"size"` // <-- Size's specDescriptors.path is inferred from this JSON tag.
+	}
+	```
+
+3. Let the SDK infer all unmarked paths on a field for a `customresourcedefinitions.specDescriptors` entry:
+
+	```go
+	type MemcachedSpec struct {
+		// Size is the size of the memcached deployment.
+		// +operator-sdk:csv:customresourcedefinitions:type=spec
+		Size int32 `json:"size"`
+	}
+	```
+
+	The SDK uses the `Size` fields' `json` tag name as `path`, `Size` as `displayName`, and field comments as `description`.
+
+4. A comprehensive example:
+	- Infer `path`, `description`, `displayName`, and `x-descriptors` for `specDescriptors` and `statusDescriptors` entries.
+	- Create three `resources` entries each with `kind`, `version`, and `name` values.
+
+	```go
+	// Represents a cluster of Memcached apps
+	// +operator-sdk:csv:customresourcedefinitions:displayName="Memcached App",resources={{Pod,v1,memcached-runner},{Deployment,v1,memcached-deployment}}
+	type Memcached struct {
+		metav1.TypeMeta   `json:",inline"`
+		metav1.ObjectMeta `json:"metadata,omitempty"`
+
+		Spec   MemcachedSpec   `json:"spec,omitempty"`
+		Status MemcachedStatus `json:"status,omitempty"`
+	}
+
+	type MemcachedSpec struct {
+		Pods MemcachedPods `json:"pods"`
+	}
+
+	type MemcachedStatus struct {
+		Pods 		 MemcachedPods `json:"podStatuses"`
+		// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="Pod Count",xDescriptors="urn:alm:descriptor:com.tectonic.ui:podCount"
+		PodCount int 					 `json:"podCount"`
+	}
+
+	type MemcachedPods struct {
+		// Size is the size of the memcached deployment.
+		// +operator-sdk:csv:customresourcedefinitions:type=spec
+		// +operator-sdk:csv:customresourcedefinitions.type=status
+		Size int32 `json:"size"`
+	}
+	```
+
+	The generated `customresourcedefinitions` will look like:
+
+	```yaml
+	customresourcedefinitions:
+	  owned:
+	  - description: Represents a cluster of Memcached apps
+	    displayName: Memcached App
+	    kind: Memcached
+	    name: memcacheds.cache.example.com
+	    version: v1alpha1
+	    resources:
+	    - kind: Deployment
+	      name: memcached-deployment
+	      version: v1
+	    - kind: Pod
+	      name: memcached-runner
+	      version: v1
+	    specDescriptors:
+	    - description: The desired number of member Pods for the deployment.
+	      displayName: Size
+	      path: pods.size
+	    statusDescriptors:
+	    - description: The desired number of member Pods for the deployment.
+	      displayName: Size
+	      path: podStatuses.size
+	    - displayName: Size
+	      path: podCount
+	      x-descriptors:
+	      - 'urn:alm:descriptor:com.tectonic.ui:podCount'
```

Comment:
Due to the fact that there no `podCount` statusDescriptors (see statusDescriptors in [here](https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/components/descriptors/reference/reference.md#statusdescriptors)), could we use something else instead? Maybe using [`text`](https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/components/descriptors/reference/reference.md#4-text) as the example?

Reply:
Interesting, @tlwu2013 Is that the definitive list of the supported spec/status descriptors?

@estroz I think we might be linking to a slightly outdated (if largely similar) list in our docs:
https://sdk.operatorframework.io/docs/golang/references/markers/#customresourcedefinitions
https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/components/descriptors/types.ts#L3-L39



@tlwu2013 Although I've linked to the console master branch above and it lists [`podCount`](https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/components/descriptors/types.ts#L32) as one of the status capabilities.

Reply:
The link has been updated in this doc to point to the master's reference. I will update this example's `x-descriptors`.

Reply:
You're actually right @hasbro17, my bad. After tried out a bit and noticed console currently does support `podCount` as one of statusDescriptors. It's just not being used in any of the community Operators hence I am not aware of it.